pkg/occlient/templates.go

```diff
@@ -139,6 +139,51 @@ func generateSupervisordDeploymentConfig(commonObjectMeta metav1.ObjectMeta, com
 	return dc
 }
 
+func updateSupervisorDeploymentConfig(existingDc *appsv1.DeploymentConfig, commonObjectMeta metav1.ObjectMeta, commonImageMeta CommonImageMeta,
+	envVar []corev1.EnvVar, envFrom []corev1.EnvFromSource, resourceRequirements *corev1.ResourceRequirements) appsv1.DeploymentConfig {
+
+	dc := *existingDc
+
+	dc.ObjectMeta = commonObjectMeta
+	dc.Spec.Strategy.Type = appsv1.DeploymentStrategyTypeRecreate
+
+	dc.Spec.Selector = map[string]string{
+		"deploymentconfig": commonObjectMeta.Name,
+	}
+
+	dc.Spec.Template.ObjectMeta = metav1.ObjectMeta{
+		Labels: map[string]string{
+			"deploymentconfig": commonObjectMeta.Name,
+		},
+		// https://github.com/openshift/odo/pull/622#issuecomment-413410736
+		Annotations: map[string]string{
+			"alpha.image.policy.openshift.io/resolve-names": "*",
```

Comment:
this looks like a duplication of what is already in `occlient.generateSupervisordDeploymentConfig` (templates.go)